### ISSUE TITLE
Wikidata: フェイスブックID・公式フェイスブックページの登録方法

### ISSUE THREAD

**Issue by takahashim** [codeforjapan/codeforelection#72]
#61 に関連するのですが、分けた方がわかりやすそうなので別issueにしてみます。

Facebookに関して、Wikidataでは2つのpropertyがあるようです。

* フェイスブックID(Facebook profile ID): P2013 https://www.wikidata.org/wiki/Property:P2013
* 公式Facebookページ(official Facebook page): P4003 https://www.wikidata.org/wiki/Property:P4003

これを見る限りでは、Facebookページについては「フェイスブックID」ではなく「公式Facebookページ」に登録するべきなのかな、と素直に思います。が、説明のところには「official Facebook page of this entity (only for use with URLs containing "/pages/")」と書いてあり、また「書式指定URL」でも「 https://www.facebook.com/pages/$1 」と書いてあるのでした。これには従わないとまずそうです。
しかし、残念ながらこのフォーマットになっているのは（フェイスブックIDの欄に）2つあるだけで、それ以外はいわゆるFacebookページであってもこんなURLにはなってないわけです。

というわけで表題に戻るのですが、現状のGrayDBの2つのカラムについて、Wikidataにはどのように登録するのがよいでしょうか。

**Comment by hkwi** [comment]
なーるほどFacebookのURLスキームの変遷が垣間見えて面白いですねぇ。たぶんこういう理由なんだろうというのは、なんとなく予想はつくので、どこかに雑文で書くとして、私の提案としては[議論](https://www.wikidata.org/wiki/Wikidata:Property_proposal/Archive/32#P2013)を遡って、`P553` or `P554` に先祖返りさせるのはどうでしょうか？
あと GrayDB のカラムはもはや当てにならないので、（分類が必要であるなら）いっそのこと両方一度混ぜてから、機械的な自動判別に任せたほうが簡単かと思います。

**Comment by takahashim** [comment]
P554というと https://www.wikidata.org/wiki/Property:P554 ですね。
ただ、こちらも上記を見るとFacebookページ的なものというよりは「username on a website」とか言っているので、大丈夫かな…というのがあります。

値に関して当てにならないというか精度が低いのはその通りのようなので、登録情報は別途用意した方がよいかもしれません。

**Comment by takahashim** [comment]
どっちかというと、P856(公式ウェブサイト)にP553的なものを組み合わせられればいいんですけど、そういう使い方は想定されてないみたいなのが厳しそうです。

**Comment by hkwi** [comment]
取りえず入る分だけは入れるという方針で行くなら、profile になっていて、かつリダイレクト先が存在するもの（つまり `/profile.php` でないもの）を抜き出して、`P2013` に入れてみましょうか。差分生成をそれで仕込んでみます。

**Comment by takahashim** [comment]
そうですね、それに加えて、 https://www.wikidata.org/wiki/Property_talk:P2013 によれば、
 `/profile.php?=XXX` となっているものについては、`XXX` で登録することになっているようです。

**Comment by hkwi** [comment]
なるほどなるほど。現在では `P2013` は意味が拡張されて、当初人物限定だったのが、人物、組織、イベントなどに対応付け可能と考えるべき、と解釈を変更する方向で行く感じなんですね。確かに `/{NUMERIC}` は `/profile.php?id={NUMERIC}` にリダイレクトされるようになっているので、実用上問題なさそうですね。

**Comment by takahashim** [comment]
フェイスブックIDについてはWikidataに反映済みです。
公式フェイスブックページについては、 https://www.wikidata.org/wiki/Property:P4003 の「official Facebook page of this entity (only for use with URLs containing "/pages/")」に合致しないURLが多すぎるので今のところ手付かずです。
いい登録方法が見つかるか、あるいはいっそ公式フェイスブックページはWikidataに登録しなくてもいいか、ということに決まればこのissueは閉じてしまってもよいかと思います。

**Comment by nyampire** [comment]
Wikidata登録ガイドとして、Wikidataコミュニティにフィードバックする
